File: Java/lab12/src/main/java/org/example/PhoneOrderFactory.java
package org.example;

import java.util.HashMap;
import java.util.Map;

public class PhoneOrderFactory {
    private final Map<String, PhoneOrder> phoneMap=new HashMap<>();
     public PhoneOrder getPhoneOrderType(String typePhoneOrder){
         PhoneOrder phoneOrder =phoneMap.get(typePhoneOrder);
         if(phoneOrder ==null){
             phoneOrder =new PhoneOrder(typePhoneOrder);
             System.out.println("Creating type of phone order like "+ phoneOrder.name);
             phoneMap.put(typePhoneOrder, phoneOrder);
         }
         return phoneOrder;
     }
}

File: Python/lab12/PhoneOrderFactory.py
from PhoneOrder import PhoneOrder

class PhoneOrderFactory:
    def __init__(self):
        self.phone_map = {}

    def get_phone_order_type(self, type_phone_order):
        phone_order = self.phone_map.get(type_phone_order)
        if phone_order is None:
            phone_order = PhoneOrder(type_phone_order)
            print("Creating type of phone order like", phone_order.name)
            self.phone_map[type_phone_order] = phone_order
        return phone_order
